### ISSUE TITLE
Present "Relevant" context files

### ISSUE THREAD

**Issue by dominiccooney** [sourcegraph/sourcegraph#52154]
TL;DR the "context files" widget started life as a way to disclose what was being sent to the LLM. But it turns out there's useful files in there, so we linkified all of the files. Now we're wondering if it can be made even more relevant and useful. Maybe the LLM could indicate which ones are particularly relevant and we could surface these more prominently.

This idea comes from this thread: https://sourcegraph.slack.com/archives/C053L1AQ0BC/p1684382337955409?thread_ts=1684366867.046149&cid=C053L1AQ0BC

#### Steps to reproduce:

1. Ask Cody something where it is given files as context

#### Expected behavior:

The presentation of the files could be smarter—can the LLM indicate, among the context items we gave it, which ones were relevant so we can make these more prominent? Maybe outside of the closed disclosure widget?

#### Actual behavior:

You get a list of all context files, which is good for disclosing which ones were provided to the LLM. But there are sometimes marginally useful results mixed in there.
